cryostatio/cryostat-web
# feat(dashboard): first start dashboard configuration

Signed-off-by: Max Cao <macao@redhat.com>

# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
_______________________________________________

Fixes: #866 

## Description of the change:
Change adds a default configuration to the dashboard for new users if their local storage `DASHBOARD_CFG` key does not exist. This works because a new user to v2.3.0 will not have this key set and a default dashboard will appear. Any other modification to the dashboard after that does not remove the key entirely, only modifies it, unless the user modifies their local storage themselves. 

A new card is added called the `QuickStart` Card. This will be iterated on in the tutorial PR for this issue https://github.com/cryostatio/cryostat-web/issues/880. Maybe tutorial steps can be done on this welcome card as an example.

Localization of the Quick Starts can come in a future separate patch if this idea is OK'd.

## How to manually test:
1. Run the pr normally with the github action.
2. Remove ALL cards in your dashboard if any. Once there are no elements in the `DASHBOARD_CFG` local storage json key, delete the key.
3. Refresh the page, there should be a QuickStart card automatically added. This should happen to new users, since they should not have any configuration in their localstorage on new releases.
4. You can also add a QuickStart card in the AddCard.
5. These QuickStart cards are not meant to be finished, they are only a UI proposal. More on tutorials will be finished in https://github.com/cryostatio/cryostat-web/issues/880
